livekit-agents/livekit/agents/beta/workflows/warm_transfer.py

```diff
@@ -0,0 +1,352 @@
+so that the human agent could gain sufficient knowledge to help the caller directly.
+
+# Context
+
+In the conversation, user refers to the human agent, caller refers to the person who's transcript is included.
+Remember, you are not speaking to the caller right now, you are speaking to the human agent.
+
+Once the human agent has confirmed, you should call the tool `connect_to_caller` to connect them to the caller.
+
+Start by giving them a summary of the conversation so far, and answer any questions they might have.
+
+## Conversation history with caller
+{conversation_history}
+## End of conversation history with caller
+
+You are talking to the human agent now,
+give a brief introduction of the conversation so far, and ask if they want to connect to the caller.
+"""
+
+
+@dataclass
+class WarmTransferResult:
+    human_agent_identity: str
+
+
+class WarmTransferTask(AgentTask[WarmTransferResult]):
+    def __init__(
+        self,
+        target_phone_number: str,
+        *,
+        hold_audio: NotGivenOr[AudioSource | AudioConfig | list[AudioConfig] | None] = NOT_GIVEN,
+        sip_trunk_id: NotGivenOr[str] = NOT_GIVEN,
+        extra_instructions: str = "",
+        chat_ctx: NotGivenOr[llm.ChatContext] = NOT_GIVEN,
+        turn_detection: NotGivenOr[TurnDetectionMode | None] = NOT_GIVEN,
+        tools: NotGivenOr[list[llm.FunctionTool | llm.RawFunctionTool]] = NOT_GIVEN,
+        stt: NotGivenOr[stt.STT | None] = NOT_GIVEN,
+        vad: NotGivenOr[vad.VAD | None] = NOT_GIVEN,
+        llm: NotGivenOr[llm.LLM | llm.RealtimeModel | None] = NOT_GIVEN,
+        tts: NotGivenOr[tts.TTS | None] = NOT_GIVEN,
+        allow_interruptions: NotGivenOr[bool] = NOT_GIVEN,
+    ) -> None:
+        super().__init__(
+            instructions=self.get_instructions(
+                chat_ctx=chat_ctx, extra_instructions=extra_instructions
+            ),
+            chat_ctx=NOT_GIVEN,  # don't pass the chat_ctx
+            turn_detection=turn_detection,
+            tools=tools or [],
+            stt=stt,
+            vad=vad,
+            llm=llm,
+            tts=tts,
+            allow_interruptions=allow_interruptions,
+        )
+
+        self._caller_room: rtc.Room | None = None
+        self._human_agent_sess: AgentSession | None = None
+        self._human_agent_failed_fut: asyncio.Future[None] = asyncio.Future()
+        self._human_agent_identity = "human-agent-sip"
+
+        self._target_phone_number = target_phone_number
+        self._sip_trunk_id = (
+            sip_trunk_id if is_given(sip_trunk_id) else os.getenv("LIVEKIT_SIP_OUTBOUND_TRUNK", "")
+        )
+        if not self._sip_trunk_id:
+            raise ValueError(
+                "`LIVEKIT_SIP_OUTBOUND_TRUNK` environment variable or `sip_trunk_id` argument must be set"
+            )
+
+        # background audio and io
+        self._background_audio = BackgroundAudioPlayer()
```

Comment:
I never really tested that, but do we know if two BackgroundAudio could conflict between each other?